### DISCUSSION TITLE
How can we show avalonia window in wpf project(Technical Question )

### DISCUSSION THREAD

**Discussion by heartacker** [AvaloniaUI/Avalonia#13212] [Q&A]
hi, avalonia team guys, I am an newer wpf developer,and after some discover,I think we love avalonia more,but our WPF project is little large ，so now we want to use avalonia to develop new UI IN  OUR PROJECT 

I try to create a AVALONIA window in WPf project and show them， There is a error 

Can anybody tell us how to do this we  are really really love avalonia  and we need this

**Comment by kyuranger** [comment]
Why not use Avaloina directly but show it in WPF project? If it is an old project, you should convert it to Avalonia by Avalonia XPF or rewrite it all by Avalonia completely.

**Reply by heartacker** [reply]
no ，this is company project ，we just want to ship step by step

other wise ，that  will be  a large project

**Reply by kyuranger** [reply]
If it is a company project, I suggest that it would be more suitable to migrate through Avalonia XPF. I believe others in the community think so too.

**Reply by timunie** [reply]
@heartacker if you have some budget, I'd highly suggest to reach out to the team to request an XPF License. It will offer you WPF running on other devices, while you can migrate step by step your App to be fully Avalonia. 

More: https://avaloniaui.net/XPF

**Comment by thevortexcloud** [comment]
As long as you are not doing a lot of fairly custom theming or directly coupled to WPF specific APIs, it should be fairly easy to migrate from WPF to Avalonia. I personally have done it in an afternoon with smaller projects with almost no code changes. For very large projects it's still mostly straight forwards (assuming your code is structured properly with MVVM) but it can take awhile. It mostly consists of Find and replacing things, adjusting name spaces, and using what ever the equivalent APIs are. If you are relying on things like data triggers, which don't have a built in equivalent then it can get a bit trickier. But there are workarounds for that. EG style class bindings, or using [this](https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors) package.

In any case the reverse situation (meaning running WPF inside Avalonia) is apparently possible. See [here](https://github.com/AvaloniaUI/Avalonia/pull/11488). But it does have some problems.
